.github/workflows/wheels-pure.yml

```diff
@@ -0,0 +1,192 @@
+        required: false
+        type: string
+        default: ''
+
+      test-checkout-src:
+        required: false
+        type: string
+        default: 'true'
+      test-extras:
+        required: false
+        type: string
+        default: ''
+      test-unittest:
+        required: false
+        type: string
+        default: ''
+      test-before:
+        required: false
+        type: string
+        default: 'true'
+
+jobs:
+  wheel-epoch-timestamp:
+    name: wheel epoch timestamper
+    runs-on: ubuntu-latest
+    outputs:
+      rapids_epoch_timestamp: ${{ steps.get_epoch_time.outputs.rapids_epoch_timestamp }}
+
+    steps:
+    - name: Get epoch timestamp to use as nightly version
+      id: get_epoch_time
+      run: echo "RAPIDS_EPOCH_TIMESTAMP=$(date +%s)" >> "${GITHUB_OUTPUT}"
+
+  wheel-build:
+    name: wheel build pure
+    needs: wheel-epoch-timestamp
+    runs-on: [self-hosted, linux, amd64, cpu4]
+    container:
+      image: "rapidsai/cibuildwheel_x86_64"
+      env:
+        RAPIDS_PY_WHEEL_CUDA_SUFFIX: ${{ inputs.python-package-cuda-suffix }}
+
+    outputs:
+      rapids_repository: ${{ steps.get_repo_info.outputs.rapids_repository }}
+      rapids_sha: ${{ steps.get_repo_info.outputs.rapids_sha }}
+      rapids_ref_name: ${{ steps.get_repo_info.outputs.rapids_ref_name }}
+      rapids_nightly_date: ${{ steps.get_repo_info.outputs.rapids_nightly_date }}
+
+    steps:
+    - name: checkout code repo
+      uses: actions/checkout@v3
+      with:
+        repository: ${{ inputs.repo }}
+        ref: ${{ inputs.sha }}
+        fetch-depth: 0 # unshallow fetch for setuptools-scm
+        persist-credentials: false
+
+    - name: Standardize repository information
+      id: get_repo_info
+      run: |
+        echo "RAPIDS_REPOSITORY=${{ inputs.repo || github.repository }}" | tee --append "${GITHUB_ENV}" "${GITHUB_OUTPUT}"
+        echo "RAPIDS_SHA=$(git rev-parse HEAD)" | tee --append "${GITHUB_ENV}" "${GITHUB_OUTPUT}"
+        echo "RAPIDS_REF_NAME=${{ inputs.branch || github.ref_name }}" | tee --append "${GITHUB_ENV}" "${GITHUB_OUTPUT}"
+        echo "RAPIDS_NIGHTLY_DATE=${{ inputs.date }}" | tee --append "${GITHUB_ENV}" "${GITHUB_OUTPUT}"
+
+    - name: Build pure python wheel with pip wheel
+      run: |
+        versioneer_override="${{ inputs.python-package-versioneer-override }}"
+
+        if [ "${versioneer_override}" == "" ]; then
+          versioneer_override="$(rapids-pip-wheel-version ${{ needs.wheel-epoch-timestamp.outputs.rapids_epoch_timestamp }})"
+        fi
+          
+        export RAPIDS_PY_WHEEL_VERSIONEER_OVERRIDE="${versioneer_override}"
+        python -m pip wheel -w ./dist ${{ inputs.package-dir }} --no-deps
+      shell: bash
+
+    - name: Upload wheels to downloads.rapids.ai
+      run: rapids-upload-wheels-to-s3 dist
+      env:
+        RAPIDS_PY_WHEEL_NAME: ${{ inputs.package-name }}
+        RAPIDS_BUILD_TYPE: ${{ inputs.build-type }}
+        RAPIDS_PY_VERSION: "3.8"
```

Comment:
Yeah, so - for pure Python wheels, the Python version is irrelevant. But to make gha-tools happy, I still pass in the version of Python used to build the pure Python wheel;

So, whether or not we used python 3.X, the wheel will be `py3-none` (indicating any Python 3, any ABI - as opposed to, say `cp38-cp38` or `cp39-cp39` which is specifically the CPython implementation of that major version).

The 3.8 here just uses the lower of the two Python versions in the matrix ["3.8", "3.9"] to create a general Python 3 wheel.

Reply:
In the future, I'd like to start using the python abi, the same as wheel names as per PIP 427 for pip wheels, so this would be "py3_none" in this case, which is a more appropriate name.

However, it turns out "3.8" is load bearing, used to set the python version in a pyenv command in the test script: 
https://github.com/rapidsai/cibuildwheel-imgs/blob/main/context-citestwheel/citestwheel.sh#L21-L23

Reply:
I'll mark it for the future to improve this naming scheme (decouple the tarball name in the s3 path from the `pyenv init ${python_version}` parameter.